tests/integration/api/istiorevisiontag_test.go

```diff
@@ -0,0 +1,217 @@
+//go:build integration
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"time"
+
+	"github.com/istio-ecosystem/sail-operator/api/v1alpha1"
+	. "github.com/istio-ecosystem/sail-operator/pkg/test/util/ginkgo"
+	"github.com/istio-ecosystem/sail-operator/pkg/test/util/supportedversion"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"istio.io/istio/pkg/ptr"
+)
+
+var _ = Describe("IstioRevisionTag resource", Ordered, func() {
```

Comment:
Do we want to cover also other injection cases like pod level labeling and other combinations defined in the MutatingWebhookConfiguration? Also some negative cases e.g.:
- user creates `default` Istio CR (InPlace strategy) and then creates `default` IstioRevisionTag -> it should fail with reasonable error
- what about cases when the user is using `istioctl` to create tags?  Do we want to prevent that?

Reply:
* I'll add a case for pod labelling
* also good point about default revision vs revisiontag, I'll cover that
* I'm not sure we can prevent istioctl usage